pepys_import/core/store/data_store.py

```diff
@@ -590,8 +589,43 @@ def search_table_type(self, table_type_id):
 
     #############################################################
     # New methods
+    def find_datafile(self, datafile_name):
+        """
+        This method tries to find a Datafile entity with the given datafile_name. If it
+        finds, it returns the entity. If it is not found, it searches synonyms.
 
-    def get_datafile(self, datafile_name, datafile_type):
+        :param datafile_name:  Name of Datafile
+        :type datafile_name: String
+        :return:
+        """
+        datafile = (
+            self.session.query(self.db_classes.Datafile)
+            .filter(self.db_classes.Datafile.reference == datafile_name)
+            .first()
+        )
+        if datafile:
+            return datafile
+
+        synonym = (
```

Comment:
Hmm, since we'll be looking for synonyms for multiple table types (esp Platforms & Sensors) - I wonder if it's worth refactoring this next block into a reusable function?

Reply:
Yes, I think it is worth because it already duplicates codes. I created a new method named **synonym_search** for DataStore:

https://github.com/debrief/pepys-import/pull/73/commits/972c963728d1d0f967d751729ae83eee2dbd54d9